grpc_attestation/src/client.rs

```diff
@@ -0,0 +1,181 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::proto::{
+    streaming_session_client::StreamingSessionClient, StreamingRequest, StreamingResponse,
+};
+use anyhow::Context;
+use oak_remote_attestation::handshaker::{
+    AttestationBehavior, ClientHandshaker, Encryptor, ServerIdentityVerifier,
+};
+use tokio::sync::mpsc::Sender;
+use tonic::{transport::Channel, Request, Streaming};
+
+const MESSAGE_BUFFER_SIZE: usize = 1;
+
+/// Convenience structure for sending requests to and receiving responses from the server.
+struct GrpcChannel {
+    sender: Sender<StreamingRequest>,
+    response_stream: Streaming<StreamingResponse>,
+}
+
+impl GrpcChannel {
+    async fn create(uri: &str) -> anyhow::Result<Self> {
+        let channel = Channel::from_shared(uri.to_string())
+            .context("Couldn't create gRPC channel")?
+            .connect()
+            .await
+            .context("Couldn't connect via gRPC channel")?;
+        let mut client = StreamingSessionClient::new(channel);
+
+        let (sender, mut receiver) = tokio::sync::mpsc::channel(MESSAGE_BUFFER_SIZE);
+
+        let request_stream = async_stream::stream! {
+            while let Some(message) = receiver.recv().await {
+                yield message;
+            }
+        };
```

Comment:
Isn't there some more functional way of getting a stream from a channel? see https://docs.rs/tokio-stream/0.1.8/tokio_stream/wrappers/struct.ReceiverStream.html